turbine/src/xdp.rs

```diff
@@ -0,0 +1,167 @@
+// re-export since this is needed at validator startup
+pub use agave_xdp::set_cpu_affinity;
+#[cfg(target_os = "linux")]
+use {
+    agave_xdp::{
+        device::{NetworkDevice, QueueId},
+        load_xdp_program,
+        tx_loop::tx_loop,
+    },
+    crossbeam_channel::TryRecvError,
+    std::{sync::Arc, thread::Builder, time::Duration},
+};
+use {
+    crossbeam_channel::{Sender, TrySendError},
+    solana_ledger::shred,
+    std::{error::Error, net::SocketAddr, thread},
+};
+
+#[derive(Clone, Debug)]
+pub struct XdpConfig {
+    pub interface: Option<String>,
+    pub cpus: Vec<usize>,
+    pub zero_copy: bool,
+    pub channel_cap: usize,
+}
+
+impl XdpConfig {
+    // A nice round number
+    const DEFAULT_CHANNEL_CAP: usize = 1_000_000;
```

Comment:
We would want some monitoring set up for the bounded channels to make sure we are not overflowing them

Reply:
Now num_addrs_failed gets incremented when the channel is full

Reply:
This is 1M packets at 1 KB a piece, i.e. 1 GB of RAM worth of buffers. Is it not a bit too much? Arguably, this should never ever exceed ~100K packets (and even that is probably too much). 1M packets is far too many to hold in what effectively is a socket buffer. 
Also suggest giving it a more descriptive name that indicates this is in packets and not in bytes.